promptulate/tools/file/tools.py

```diff
@@ -0,0 +1,256 @@
+import os
+import shutil
+
+from promptulate.tools.base import Tool
+
+
+class WriteFileTool(Tool):
+    name: str = "write-file"
+    description: str = (
+        "Write to a file"
+        "Useful when you need to edit/create and edit a file."
+        "It can edit the specified file in the specified directory/local directory"
+        "If the file does not exist, edit it after it is created."
+        "The output parameters must be:"
+        "file_name(file name), text(related information)."
+    )
+
+    def __init__(self, root_dir: str, *args, **kwargs) -> None:
+        self.root_dir = root_dir
+        super().__init__(*args, **kwargs)
+
+    def _run(self, file_name: str, text: str) -> str:
+        """Write to a file
+
+        Args:
+            query: file_name and text.
+            file_name(Optional(str)): The name of the file you want to edit.
+            text(Optional(str)): The content you want to edit.
+
+        Returns:
+            str: Write file successfully
+        """
+        try:
+            with open(f"{self.root_dir}/{file_name}", "w") as file:
+                file.write(text)
+        except IOError:
+            return "An error occurred while trying to write to the file."
+        except PermissionError:
+            return "Permission denied: You don't have permission to write to this file."
+        return "Write file successfully"
+
+
+class AppendFileTool(Tool):
+    name: str = "append-file"
+    description: str = (
+        "Append to a file"
+        "Useful when you need to edit/create edit a file."
+        "It can edit the specified file in the specified directory/local directory,"
+        "Append the content to the end of the file."
+        "If the file does not exist, edit it after it is created"
+        "The output parameters must be:"
+        "file_name(file name), text(related information)."
+    )
+
+    def __init__(self, root_dir: str, *args, **kwargs) -> None:
+        self.root_dir = root_dir
+        super().__init__(*args, **kwargs)
+
+    def _run(self, file_name: str, text: str) -> str:
+        """Append to a file
+
+        Args:
+            query: file_name and text.
```

Comment:
useless parameter

Reply:
use funtion implement , that agent can konw the args.

Reply:
@ruanrongman He can add Parameters to show args. For pne's tool, maybe class type tool is better. For custom tool, it's more simple for developer.

Reply:
> @ruanrongman He can add Parameters to show args. For pne's tool, maybe class type tool is better. For custom tool, it's more simple for developer.

ok,class type is better